### ISSUE TITLE
Simple example failure

### ISSUE THREAD

**Issue by zodo** [scalalandio/chimney#40]
Hello
```
scala> case class One(text: Option[String])
scala> case class Two(text: Option[String])
scala> val one = One(None)
scala> import io.scalaland.chimney.dsl._
scala> one.transformInto[Two]
<console> error: could not find implicit value for parameter derivedTransformer: io.scalaland.chimney.DerivedTransformer[One,Two,shapeless.HNil]
             one.transformInto[Two]
                              ^
```
Scala version: 2.12.3,
Library version: 0.1.5
Any advice?

**Comment by krzemin** [comment]
@zodo thanks for the bug report. There were ambiguous transformer instances for this specific case. Fixed in #41.

**Comment by krzemin** [comment]
There were similar issues concerning other types that we provide particular instances for (like either, collections, maps). All of them were addressed in #41.

Closing, as it's now merged.
